api/app/services/escalation/escalation_event_broker.py

```
@@ -0,0 +1,49 @@
+    ) -> AsyncIterator[asyncio.Queue[Escalation]]:
+        queue: asyncio.Queue[Escalation] = asyncio.Queue(maxsize=1)
+        async with self._lock:
+            self._subscribers[message_id].add(queue)
+
+        try:
+            yield queue
+        finally:
+            async with self._lock:
+                queues = self._subscribers.get(message_id)
+                if not queues:
+                    return
```

Ruff (0.15.15)

[warning] `return` inside `finally` blocks cause exceptions to be silenced

(B012)